core/XH.ts

```diff
@@ -484,7 +484,7 @@ export class XHApi {
      *
      * @returns value of input if user confirms, false if user cancels.
      */
-    prompt(config: MessageSpec): Promise<any> {
```

Comment:
1)  As part of this change I think message should also get a generic return type of `<T | boolean>` 
2) I think that T should default to `unknown`. does that work.

Reply:
Had the same thought about defaulting to `unknown`.  Doesn't really provide a whole lot (most likely you'll need to specify the type to actually work with the returned data), but I think it makes sense.

Reply:
I think that defaulting to unknown encourages the user to use this type feature, which is nice because then it reminds the user to properly handle the `false` case in addition to their specified `T` case.
```
// myMessage is of type <string | false>
const myMessage = await XH.prompt<string>({
    ...
});

if (!myMessage) return;

// myMessage is now considered to be of type <string> by IntelliJ
```

Reply:
How this looks in action:

<img width="610" alt="Screenshot 2023-05-01 at 11 45 19 AM" src="https://user-images.githubusercontent.com/101586582/235509358-bf9e30fa-708a-489c-bfeb-7d2f4641fc98.png">

<img width="520" alt="Screenshot 2023-05-01 at 11 46 27 AM" src="https://user-images.githubusercontent.com/101586582/235509498-dbff4d67-6c7b-4d25-9d5e-e71175ce62a4.png">

Reply:
@jskupsik were you planning to push a commit with this change so we can close this out? Any more decision/discussion items here?

Reply:
just having one additional thought about this.  Given that the input defaults to a TextInput, seems like it would nice to have the type of this thing actually default to 'string'.   That's what you will have most often

Seems like a case where the typescript to do that would be kind of nasty, not sure if it is worth it. 

Also, we could probably add that intelligence at any time.

Reply:
went ahead and added the `unknown` default that we had agreed on